sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py

```diff
@@ -280,3 +280,22 @@ def _validate_component(self, deployment: Deployment, orchestrators: OperationOr
                 name, _ = parse_prefixed_name_version(deployment.job_definition.component)
                 deployment.job_definition.name = name
             deployment.job_definition.component_id = component_id
+        elif(deployment.job_definition.job):
+            if isinstance(deployment.job_definition.job, str):
+                component = PipelineComponent(source_job_id= deployment.job_definition.job)
+                rest_component = self._component_operations.create_or_update(
+                    name = component.name,
+                    resource_group_name=self._resource_group_name,
+                    workspace_name=self._workspace_name,
+                    body = component._to_rest_object(),
+                    version= component.version,
+                    **self._init_kwargs
+                )
+                deployment.job_definition.job = None
+                deployment.job_definition.component_id = rest_component.id
+                if not deployment.job_definition.name and rest_component.name:
```

Comment:
Is there a way to avoid the code duplication here?
296-301 lines are similar as 269-273